boa_engine/src/bytecompiler/mod.rs

```diff
@@ -1443,12 +1453,164 @@ impl<'b> ByteCompiler<'b> {
                     self.emit_opcode(Opcode::PushNewTarget);
                 }
             }
+            Expression::Optional(opt) => {
+                self.compile_optional_preserve_this(opt)?;
+
+                self.emit_opcode(Opcode::Swap);
+                self.emit_opcode(Opcode::Pop);
+
+                if !use_expr {
+                    self.emit_opcode(Opcode::Pop);
+                }
+            }
             // TODO: try to remove this variant somehow
             Expression::FormalParameterList(_) => unreachable!(),
         }
         Ok(())
     }
 
+    fn compile_access_preserve_this(&mut self, access: &PropertyAccess) -> JsResult<()> {
```

Comment:
I know that we are missing documentation for most of the existing functions like this one, but would it be possible to add documentations for the new ones at least, so that #1608 doesn't grow further?

Reply:
Added docs